### PR TITLE
Add lifecycle management requirements

### PR DESCRIPTION
**Pull request by seroyer**
Signed-off-by: Steven Royer <sroyer@redhat.com>

### CODE REVIEW

File: LIFECYCLE.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Lifecycle Management
+
+## Device Lifecycle scope
+
+High level requirements on a DPU/IPU or subsequently a "device"
+
+* a way to reboot a device
+  * if connected to a host, the host should be able to survive a device reset
```
Comment (glimchb) [inline LIFECYCLE.md:8]: here you can link to BOOTSEQ.md

Comment (jainvipin) [reply LIFECYCLE.md:8]: Clearly a device that is exposing PCIe to the host cant' reset fully.
What you mean by device reset is perhaps update/reset of the OS (and not the entire device), or the part that needs patching (e.g. new datapath to upgrade xPU's network/storage/telemetry functions). It is worth clarifying what reset is sought after here should be very clear and tied to the rough components of an xPU that we can use a base for our discussion.

File: LIFECYCLE.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+* a way to update a device
+  * firmware update
+  * OS update
+  * software/application update
+* a way to recover a device
+  * reset to a known good state, e.g., factory reset
```
Comment (jainvipin) [inline LIFECYCLE.md:14]: Recovery from over the network, or from host, or both?

File: LIFECYCLE.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+* a way to debug a device
```
Comment (jainvipin) [inline LIFECYCLE.md:15]: cover aspects to find out if the device crashed, how to debug, recover, and provide post-mortem information
how to debug problems, etc.
More importantly - is there a desire to do this uniformly across various xPUs?

**Review by jainvipin — COMMENTED** [review]
there are details to discuss here from requirements point of view. But without a functional/architectural block diagram that various xPU vendors agree to, these discussions will not be grounded.

File: LIFECYCLE.md (hunk 1 above)
Comment (glimchb) [reply LIFECYCLE.md:8]: sometimes the device will either reset fully or crash due to bug, we need to accommodate what happens there... 
Is PCIe DPC going to help us, for example ? Will host just crash ?

File: LIFECYCLE.md (hunk 2 above)
Comment (glimchb) [reply LIFECYCLE.md:14]: both (could be parameter in API call)

File: LIFECYCLE.md (hunk 3 above)
Comment (glimchb) [reply LIFECYCLE.md:15]: at least fetch logs or core dumps, for example are desired to be uniformed... maybe more...

Comment (seroyer) [reply LIFECYCLE.md:15]: Yes, I think there is a desire to uniformly collect failure data and recover.  Perhaps the format of that data is different, and of course the specifics of live debugging a card will be different for different hardware and software vendors.

File: LIFECYCLE.md (hunk 1 above)
Comment (seroyer) [reply LIFECYCLE.md:8]: I think this needs more discussion.  Though perhaps that discussion should be more targeted at the contents BOOTSEQ.md?  My plan is to replace this bullet with the link to the BOOTSEQ.md.

Comment (glimchb) [reply LIFECYCLE.md:8]: go for it

File: LIFECYCLE.md (hunk 2 above)
Comment (glimchb) [reply LIFECYCLE.md:14]: +1, good starting point

**Review by glimchb — APPROVED** [review]

**Pull request merged**